src/core/common/string.cpp

```diff
@@ -35,6 +35,18 @@
 
 namespace ot {
 
+uint16_t StringLength(const char *aString, uint16_t aMaxLength)
```

Comment:
Is it possible to provide as a weak implementation of `strnlen()`?

Reply:
How would you propose to keep the current unit tests for `StringLength()`? Would you define a weak strnlen that uses `StringLength()`?

Reply:
Got it!